### PR TITLE
feat: add curl config

### PR DESCRIPTION
**Pull request by zhangymPerson**

### CODE REVIEW

**Review by smeech — COMMENTED** [review]
A straightforward text replacement package.
Well documented, and `tags:` in `_manifest.yml`.
Looks safe to merge, but does present various `curl` commands, so needs double-checking/approval by someone more experienced with these than me.

**Comment by smeech** [comment]
I've done some reading around `curl` and am happy the package is pretty safe, even if `curl` itself has risks. I'll merge it subject to one issue - see Review.

It might be interesting to add a global variable, along the lines of:
```yml
global_vars:
  - name: curlform
    type: form
    params:
      layout: Target https://[[URL]]
```
and replace all instances of `example.com` with `{{curlform.URL}}` e.g.:
```yml
matches:
  - trigger: ":curl:"
    replace: "curl --request GET https://{{curlform.URL}}"
```
The other optional variables: `value`, `key`, `file/path` etc. could also be supplied by forms in a similar manner, or [cursor hints](https://espanso.org/docs/matches/basics/#cursor-hints) used to position the cursor in simpler cases.

File: packages/curl/0.1.0/package.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    replace: "curl --request PUT https://example.com"
+
+  - trigger: ":curldelete:"
+    replace: "curl --request DELETE https://example.com"
+
+  - trigger: ":curlpatch:"
+    replace: "curl --request PATCH https://example.com"
+
+  - trigger: ":curldata:"
+    replace: "curl --request POST --header 'Content-Type: multipart/form-data' --form 'key=value' http://example.com"
```
Comment (smeech) [inline packages/curl/0.1.0/package.yml:26]: Is there a reason for the mix of `https:` and `http:`? The former throughout might be safer? I'd be happy to merge if this is addressed (unless there are good reasons not to).

Comment (zhangymPerson) [reply packages/curl/0.1.0/package.yml]: Yes, I added the relevant global variables as per your prompt

**Comment by zhangymPerson** [comment]
@smeech Thank you very much for your guidance. I have learned a better way of configuration from this pr

**Pull request merged**